birdseye/server.py

```diff
@@ -28,8 +28,7 @@ class FileConverter(PathConverter):
 @app.route('/')
 def index():
     files = sorted(all_file_paths())
-    prefix = os.path.commonprefix(files)
-    files = zip(files, [strip_required_prefix(f, prefix) or path_leaf(f)
+    files = zip(files, [short_path(f)
```

Comment:
Doing this means calling `all_file_paths()` (which includes a database call) for every file, which is wasteful.

Also we talked about creating one or more functions which could be unit tested without hitting the database, and then adding those tests. Could you please do that?

Reply:
Good point, yes. With that in mind, I've tried to go back and combine the logic from index and file_view in the single function, but it still ends up cleaner for now to just to keep them separate.

I have pushed new commits, which apply a split function fix to both server.py and utils.py.

I think I understand what you mean with writing tests, but I am out of my depth with writing tests. I have not used unittest before so am struggling to make sense of the structure and how the existing tests run.